packages/reactive-element/src/reactive-element.ts

```diff
@@ -1317,32 +1317,33 @@ export abstract class ReactiveElement
   _$changeProperty(
     name: PropertyKey,
     oldValue: unknown,
-    options: PropertyDeclaration,
+    {useDefault, reflect, wrapped}: PropertyDeclaration,
     initializeValue?: unknown
   ) {
     // Record default value when useDefault is used. This allows us to
     // restore this value when the attribute is removed.
-    if (options.useDefault && !(this.__defaultValues ??= new Map()).has(name)) {
+    if (useDefault && !(this.__defaultValues ??= new Map()).has(name)) {
       this.__defaultValues.set(
         name,
         initializeValue ?? oldValue ?? this[name as keyof this]
       );
       // if this is not wrapping an accessor, it must be an initial setting
       // and in this case we do not want to record the change or reflect.
-      if (options.wrapped !== true || initializeValue !== undefined) {
+      if (wrapped !== true || initializeValue !== undefined) {
         return;
       }
     }
     // TODO (justinfagnani): Create a benchmark of Map.has() + Map.set(
     // vs just Map.set()
     if (!this._$changedProperties.has(name)) {
+      oldValue = !this.hasUpdated && !useDefault ? undefined : oldValue;
```

Comment:
I slightly prefer an if that doesn't reassign to the existing value, for readability:

```suggestion
      // On the initial change, the old value should be `undefined`, except
      // with `useDefault`
      if (!this.hasUpdated && !useDefault) {
        oldValue = undefined;
      }
```